datasets/gb/nca_enforcements/crawler.py

```diff
@@ -0,0 +1,137 @@
+from typing import List, Literal
+
+from pydantic import BaseModel, Field
+from zavod.shed.gpt import DEFAULT_MODEL, run_typed_text_prompt
+from zavod.stateful.review import (
+    HtmlSourceValue,
+    assert_all_accepted,
+    review_extraction,
+)
+
+from zavod import Context
+from zavod import helpers as h
+
+NAME_XPATH: str = "//h1[@itemprop='headline']/text()"
+CONTENT_XPATH: str = "//div[@itemprop='articleBody']"
+DATE_XPATH: str = "//div[@itemprop='articleBody']/p[strong][last()]/strong/text()"
+TOPIC_XPATH: str = "//li[@itemprop='keywords']/a/text()"
+
+Schema = Literal["Person", "Company", "LegalEntity"]
+
+schema_field = Field(
+    description="Use LegalEntity if it isn't clear whether the entity is a person or a company."
+)
+
+
+class Offender(BaseModel):
+    entity_schema: Schema = schema_field
+    name: str
+    address: List[str] = []
+    country: List[str] = []
+
+
+class Offenders(BaseModel):
+    offenders: List[Offender]
+
+
+PROMPT = f"""
+Extract the offenders or entities subject to UK National Crime Agency (NCA) enforcement actions mentioned in the article.
+NEVER include victims, witnesses, investigators, or enforcement officers.
+NEVER infer, assume, or generate values that are not directly stated in the source text.
+If the name refers to an individual person, use Person as the entity_schema.
+
+Specific fields:
+
+- name: The name of the entity precisely as expressed in the text.
+- entity_schema: {schema_field.description}
+- address: The locations associated with the offender, such as their home address, town, city, or region."
+- country: Any countries the entity is indicated to reside, operate, or have been born or registered in. Leave empty if not explicitly stated.
+"""
+
+
+def crawl_enforcement_action(context: Context, url: str) -> None:
+    article_doc = context.fetch_html(url, cache_days=7, absolute_links=True)
+    if article_doc is None:
+        return
+    article_name = article_doc.xpath(NAME_XPATH)[0]
+    assert article_name, "Article name not found"
+    article_content = article_doc.xpath(CONTENT_XPATH)
+    assert len(article_content) == 1
+    article_element = article_content[0]
+    date = article_doc.xpath(DATE_XPATH)
+    # The date is absent in some articles
+    if date != []:
+        date = date[0]
```

Comment:
just a nitpick. Since the way we query for things with xpath() gives us lists, I prefer calling the variables their plural. Then we also don't have to use the same variable with two different types (list and string)
```suggestion
    dates = article_doc.xpath(DATE_XPATH)
    # The date is absent in some articles
    if dates != []:
        date = dates[0]
```